integration-libs/opf/b2b-checkout/root/guards/opf-b2b-checkout-steps-set.guard.ts

```diff
@@ -0,0 +1,186 @@
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+import {
+  CheckoutCostCenterFacade,
+  CheckoutPaymentTypeFacade,
+} from '@spartacus/checkout/b2b/root';
+import {
+  CheckoutFlowOrchestratorService,
+  CheckoutStepService,
+  CheckoutStepsSetGuard,
+} from '@spartacus/checkout/base/components';
+import {
+  CheckoutDeliveryAddressFacade,
+  CheckoutDeliveryModesFacade,
+  CheckoutPaymentFacade,
+  CheckoutStep,
+  CheckoutStepType,
+} from '@spartacus/checkout/base/root';
+import { LoggerService, RoutingConfigService } from '@spartacus/core';
+import { combineLatest, Observable, of } from 'rxjs';
+import { filter, map, take, switchMap, tap } from 'rxjs/operators';
+import { OPF_CHECKOUT_FLOW_NAME } from '@spartacus/opf/checkout/root';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class OpfB2bCheckoutStepsSetGuard extends CheckoutStepsSetGuard {
+  protected logger = inject(LoggerService);
+  protected checkoutFlowOrchestratorService = inject(
+    CheckoutFlowOrchestratorService
+  );
+
+  constructor(
+    protected checkoutStepService: CheckoutStepService,
+    protected routingConfigService: RoutingConfigService,
+    protected checkoutDeliveryAddressFacade: CheckoutDeliveryAddressFacade,
+    protected checkoutPaymentFacade: CheckoutPaymentFacade,
+    protected checkoutDeliveryModesFacade: CheckoutDeliveryModesFacade,
+    protected router: Router,
+    protected checkoutPaymentTypeFacade: CheckoutPaymentTypeFacade,
+    protected checkoutCostCenterFacade: CheckoutCostCenterFacade,
+    protected activeCartFacade: ActiveCartFacade
+  ) {
+    super(
+      checkoutStepService,
+      routingConfigService,
+      checkoutDeliveryAddressFacade,
+      checkoutPaymentFacade,
+      checkoutDeliveryModesFacade,
+      router,
+      activeCartFacade
+    );
+  }
+
+  canActivate(route: ActivatedRouteSnapshot): Observable<GuardResult> {
+    let currentIndex = -1;
+    const currentRouteUrl = '/' + route.url.join('/');
+
+    return combineLatest([
+      this.checkoutStepService.steps$,
+      this.checkoutPaymentTypeFacade.isAccountPayment(),
+      this.isOpfCheckoutFlow(),
+    ]).pipe(
+      tap(([, isAccount, isOpf]) => {
+        this.checkoutStepService.disableEnableStep(
+          CheckoutStepType.PAYMENT_DETAILS,
+          isAccount
+        );
+        if (isOpf) {
+          this.checkoutStepService.disableEnableStep(
+            CheckoutStepType.REVIEW_ORDER,
+            !isAccount
+          );
+        }
+      }),
+      take(1),
+      switchMap(([steps, isAccount]) => {
+        currentIndex = steps.findIndex((step) => {
+          const stepRouteUrl = `/${
+            this.routingConfigService.getRouteConfig(step.routeName)?.paths?.[0]
+          }`;
+          return stepRouteUrl === currentRouteUrl;
+        });
+        // get current step
+        let currentStep;
+        if (currentIndex >= 0) {
+          currentStep = steps[currentIndex];
+        }
+        if (Boolean(currentStep)) {
```

Comment:
We don't need a boolean conversion in an if condition.

Reply:
Since we are extending from CheckoutStepsSetGuard (which already exists in the SPA codebase), I aimed to keep the original code as untouched as possible, adding only the minimal adjustments required for OPF. The part you mentioned was directly copied from the parent guard. In my view, this approach helps ensure code stability by preserving the well-tested functionality of the base guard, while also improving maintainability.